ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -256,6 +257,13 @@ var _ = Context("Inside the default namespace", func() {
 				time.Second*15, time.Millisecond*500).Should(Equal(1), fmt.Sprintf("workerGroup %v", workerPods.Items))
 		})
 
+		It("should not modify workers to delete", func() {
+			// workersToDelete persists until removed
+			Consistently(
+				getNumWorkersToDeleteFunc(ctx, myRayCluster),
+				time.Second*1, time.Millisecond*200).Should(Equal(1), fmt.Sprintf("workerGroup %v", workerPods.Items))
```

Comment:
Does the order of tests matter? To elaborate, this test will pass only if it is executed after the test "should update a raycluster object".

https://github.com/ray-project/kuberay/blob/0bd28e758fe89c69ab3e207baa09644562a393c6/ray-operator/controllers/ray/raycluster_controller_test.go#L247

 If the order matters, we can remove the dependency by the following pseudo code:
```go
numPendingDeleteWorkers = getNumWorkersToDeleteFunc(ctx, myRayCluster)
Consistently(
     getNumWorkersToDeleteFunc(ctx, myRayCluster),
     time.Second*1, time.Millisecond*200).Should(Equal(numPendingDeleteWorkers), fmt.Sprintf("workerGroup %v", workerPods.Items))
```

Reply:
Ordering matters in that I'd want to test persistence of workersToDelete after adding the workersToDelete file.
The pseudo code is a good suggestion.
For concreteness, I'd add an assertion that numPendingDeleteWorkers is 1 .

Reply:
Make sense.